exonum/src/blockchain/mod.rs

```diff
@@ -226,24 +231,28 @@ impl BlockchainMut {
     }
 
     /// Creates and commits the genesis block with the given genesis configuration.
-    // TODO: extract genesis block into separate struct [ECR-3750]
-    fn create_genesis_block(
-        &mut self,
-        config: ConsensusConfig,
-        initial_services: Vec<InstanceConfig>,
-    ) -> Result<(), Error> {
-        config.validate()?;
+    fn create_genesis_block(&mut self, genesis_config: GenesisConfig) -> Result<(), Error> {
+        genesis_config.consensus_config.validate()?;
         let mut fork = self.fork();
-        Schema::new(&fork).consensus_config_entry().set(config);
+        Schema::new(&fork)
+            .consensus_config_entry()
+            .set(genesis_config.consensus_config);
+
+        for ArtifactSpec { artifact, payload } in genesis_config.artifacts {
+            if !self.dispatcher.is_artifact_deployed(&artifact) {
+                self.dispatcher
+                    .deploy_artifact_sync(&fork, artifact, payload)?;
+            }
+        }
 
         // Add service instances.
-        for instance_config in initial_services {
-            self.dispatcher.add_builtin_service(
-                &mut fork,
-                instance_config.instance_spec,
-                instance_config.artifact_spec.unwrap_or_default(),
-                instance_config.constructor,
-            )?;
+        for InstanceInitParams {
```

Comment:
Nit: Multi-line destructuring harms readability, IMO.

Reply:
I agree, but that's what rustfmt applies to this specific part of code...